torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1530,12 +1593,21 @@ def _verify(expr, potential_expr):
             for i, ss in enumerate(t.size()):
                 property_source = TensorPropertySource(source, TensorProperty.SIZE, i)
                 track_symint(property_source, ss)
-                if _is_dim_dynamic(t, i):
-                    # If this dim is marked dynamic, we need to do a test on it, to ensure that it has not bee
-                    # constrained to an integer.
-                    if _is_int(ss):
-                        raise RuntimeError(f"Attempting to constrain dim {i} for {source}, which violates user's mark_dynamic")
-                    dynamic_sources.append(property_source)
+                if dynamic_ranges:
+                    dyn_range = dynamic_ranges[pos]
+                    if dyn_range and i in dyn_range:
```

Comment:
Why do you need to test `dyn_range`; your type annotation claims that `dyn_range` must be `MinMaxConstraint`